src/stavetext.ts

```diff
@@ -31,36 +52,41 @@ export class StaveText extends StaveModifier {
     };
   }
 
-  getCategory() {
+  getCategory(): string {
     return StaveText.CATEGORY;
   }
-  setStaveText(text) {
+
+  setStaveText(text: string): this {
     this.text = text;
     return this;
   }
-  setShiftX(x) {
+
+  setShiftX(x: number): this {
     this.shift_x = x;
     return this;
   }
-  setShiftY(y) {
+
+  setShiftY(y: number): this {
     this.shift_y = y;
     return this;
   }
 
-  setFont(font) {
-    Vex.Merge(this.font, font);
+  setFont(font: FontInfo): this {
+    this.font = font;
```

Comment:
This changes the interface because `this.font` is already initialized. So if I do `setFont({size: 12}`, then `family` and `weight` will be clobbered.

This should be `this.font = {...this.font, ...font}`.

Reply:
Now none of the attributes (`family`, `size` or `weight`) is optional. 
```js
export interface FontInfo {
  size: number;
  weight: string;
  family: string;
}
```
In order this to work we have to make at least one attibute optional. Do you want all of them optional? 
```js
export interface FontInfo {
  size?: number;
  weight?: string;
  family?: string;
}
```

Reply:
Right, I think it's good to have them as non-optional. But for JS compatibility, we don't want to change the behavior.

Reply:
Understood, done.